sqlmesh/core/engine_adapter/trino.py

```diff
@@ -58,6 +59,15 @@ def set_current_catalog(self, catalog: str) -> None:
         """Sets the catalog name of the current connection."""
         self.execute(exp.Use(this=schema_(db="information_schema", catalog=catalog)))
 
+    def get_catalog_type(self, catalog: t.Optional[str]) -> str:
+        if catalog:
+            connector_name = self.fetchone(
+                f"select connector_name from system.metadata.catalogs where catalog_name='{catalog}'"
+            )
+            if len(connector_name) > 0:
+                return connector_name[0]
```

Comment:
Catalogs in Trino need to have unique names or Trino will fail to start, so there should only ever be a single row returned.

Since `fetchone()` returns a single row as a tuple, so the length check is to see if there was a value in the first column `connector_name`, and if there is, it gets returned.

If no records are returned, we just return the default

Reply:
Ah I see now. You could then use `seq_get(connector_name, 0) or self.DEFAULT_CATALOG_TYPE` or this is fine too.

Reply:
I like that, its more concise. I added it